Source/carthage/Build.swift

```diff
@@ -163,34 +163,43 @@ public struct BuildCommand: CommandProtocol {
 		}
 	}
 
-	/// Opens a temporary file on disk, returning a handle and the URL to the
+	/// Opens an existing file, if provided, or creates a temporary file if not, returning a handle and the URL to the
 	/// file.
-	private func openLogFile(_ path: String?) -> SignalProducer<(FileHandle, URL), NSError> {
+	private func openLogFile(_ path: String?) -> SignalProducer<(FileHandle, URL), CarthageError> {
 		return SignalProducer.attempt {
-			let logFD: Int32
-			let logPath: String
-			
 			if let path = path {
-				logFD = FileHandle(forUpdatingAtPath: path)?.fileDescriptor ?? -1
-				logPath = path
+				if !FileManager.default.fileExists(atPath: path) {
+					FileManager.default.createFile(atPath: path, contents: nil, attributes: nil)
+				}
+				
+				let fileURL = URL(fileURLWithPath: path, isDirectory: false)
+				
+				guard let handle = FileHandle(forUpdatingAtPath: path) else {
```

Comment:
The documentation for [`FileHandle.init(forUpdatingAtPath:)`](https://developer.apple.com/reference/foundation/filehandle/1411131-init) tasks us with closing the file descriptor.

My inclination (with no existing `FileHandle.init(forUpdatingAtPath:closeOnDealloc:)` to call) is to do some sort of transfer from `FileHandle.init(forUpdatingAtPath:)` `flatMap`-ing `FileHandle(fileDescriptor: $0.fileDescriptor, closeOnDealloc: true)`.

Do you think (knowing the “immediately deallocated” issue you described), the following would work?
```swift
guard
	!FileManager.default.fileExists(atPath: path) &&
	FileManager.default.createFile(atPath: path, contents: nil, attributes: nil),
	case let fileURL = URL(fileURLWithPath: path, isDirectory: false),
	let handle = FileHandle(forUpdatingAtPath: path)
		.flatMap({ FileHandle(fileDescriptor: $0.fileDescriptor, closeOnDealloc: true) })
		.flatMap({ $0.fileDescriptor >= 0 ? $0 : nil })
else {
	let error = NSError(
		domain: CarthageKitBundleIdentifier,
		code: 1,
		userInfo: [NSLocalizedDescriptionKey: "Unable to open file handle for file at \(path)"]
	)
	return .failure(.writeFailed(fileURL, error))
}

return .success((handle, fileURL))
```